frontend/tests/errorHandling.test.ts

```diff
@@ -0,0 +1,152 @@
+import { expect } from '@playwright/test';
+import { test } from './fixtures';
+import { SandboxPage } from './pages/sandboxPage';
+import * as testEnv from './envVars';
+import { UserDashboardPage } from './pages/userDashboardPage';
+import { LoginPage } from './pages/loginPage';
+import { AdminDashboardPage } from './pages/adminDashboardPage';
+import { loginAs } from './authHelpers';
+import { UserAccountSettingsPage } from './pages/userAccountSettingsPage';
+import { getInbox } from './mailboxHelpers';
+
+test('can catch 500 errors from goto in same tab', async ({ page }) => {
+  await new SandboxPage(page).goto();
+  // Create promise first before triggering the action
+  const responsePromise = page.waitForResponse('/api/testing/test500NoException');
+  await page.getByText('Goto API 500', {exact: true}).click();
+  const response = await responsePromise;
+  expect(response.status()).toBe(500);
+});
+
+test('can catch 500 errors from goto in new tab', async ({ page, context }) => {
+  await new SandboxPage(page).goto();
+  const responsePromise = context.waitForEvent('response', response => response.url().endsWith('/api/testing/test500NoException'));
+  await page.getByText('Goto API 500 new tab').click();
+  const response = await responsePromise;
+  expect(response.status()).toBe(500);
+});
+
+test('can catch 500 errors in page load', async ({ page }) => {
+  await new SandboxPage(page).goto();
+  await page.getByText('Goto page load 500', {exact: true}).click();
+  // eslint-disable-next-line @typescript-eslint/quotes
```

Comment:
Wouldn't it be tidier to inverse the quotes? I.e. use singles on the outside.

Reply:
Fixed in commit a7e68c5a88cec7dadeaa21812a7153891366ab1b..